File: server/mydto/project.go
package mydto

import "github.com/uwine4850/foozy/pkg/router/rest"

type ProjectAuthor struct {
	rest.ImplementDTOMessage
	TypProjectAuthor rest.TypeId `dto:"-typeid"`
	UID              int         `dto:"UID" db:"user_id"`
	Username         string      `dto:"Username" db:"username"`
	Avatar           string      `dto:"Avatar" db:"avatar"`
}

type ProjectMessage struct {
	rest.ImplementDTOMessage
	TypProjectMessage rest.TypeId   `dto:"-typeid"`
	Author            ProjectAuthor `dto:"Author"`
	Id                int           `dto:"Id" db:"id"`
	UserId            int           `dto:"UserId" db:"user_id"`
	Name              string        `dto:"Name" db:"name"`
	Description       string        `dto:"Description" db:"description"`
	Error             string        `dto:"Error"`
}

type ProjectLogGroupMessage struct {
	rest.ImplementDTOMessage
	TypProjectLogGroupMessage rest.TypeId `dto:"-typeid"`
	Id                        int         `dto:"Id" db:"id"`
	ProjectId                 int         `dto:"ProjectId" db:"project_id"`
	Name                      string      `dto:"Name" db:"name"`
	Description               string      `dto:"Description" db:"description"`
	Error                     string      `dto:"Error"`
}

File: client/src/dto/project.ts

export interface MsgProject {
    TypProjectMessage?: unknown;
    Author: MsgProjectAuthor | undefined;
    Id: number;
    UserId: number;
    Name: string;
    Description: string;
    Error: string;
}
export function isMsgProject(obj: any): obj is MsgProject {
    return typeof obj === 'object' && obj !== null && 'TypProjectMessage' in obj;
}
export interface MsgProjectAuthor {
    TypProjectAuthor?: unknown;
    UID: number;
    Username: string;
    Avatar: string;
}
export function isMsgProjectAuthor(obj: any): obj is MsgProjectAuthor {
    return typeof obj === 'object' && obj !== null && 'TypProjectAuthor' in obj;
}
export interface MsgProjectLogGroup {
    TypProjectLogGroupMessage?: unknown;
    Id: number;
    ProjectId: number;
    Name: string;
    Description: string;
    Error: string;
    AuthorToken: string;
}
export function isMsgProjectLogGroup(obj: any): obj is MsgProjectLogGroup {
    return typeof obj === 'object' && obj !== null && 'TypProjectLogGroupMessage' in obj;
}
export interface MsgLogItem {
    TypLogItemMessage?: unknown;
    Type: number;
    Token: string;
    Error: string;
    Payload: MsgLogItemPayload | undefined;
}
export function isMsgLogItem(obj: any): obj is MsgLogItem {
    return typeof obj === 'object' && obj !== null && 'TypLogItemMessage' in obj;
}
export interface MsgLogItemPayload {
    TypLogItemPayload?: unknown;
    Id: number;
    LogGroupId: number;
    Text: string;
    Type: string;
    Tag: string;
    Datetime: string;
}
export function isMsgLogItemPayload(obj: any): obj is MsgLogItemPayload {
    return typeof obj === 'object' && obj !== null && 'TypLogItemPayload' in obj;
}
export interface MsgLogItemsFilter {
    TypLogItemsFilter?: unknown;
    Text: string;
    Type: string;
    Tag: string;
    DateTime: string;
}
export function isMsgLogItemsFilter(obj: any): obj is MsgLogItemsFilter {
    return typeof obj === 'object' && obj !== null && 'TypLogItemsFilter' in obj;
}
